syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

service Messenger {
    rpc SendMessage(ClientMessage) returns (SendMessageResponse) {}
    rpc GetAndFlushMessages(google.protobuf.Empty) returns (stream ServerMessage) {}
}

message ClientMessage {
    string author = 1;
    string text = 2;
}

message SendMessageResponse {
    google.protobuf.Timestamp sendTime = 1;
} 

message ServerMessage { 
    string author = 1;
    string text = 2;
    google.protobuf.Timestamp sendTime = 3;
}